.github/scripts/userbenchmark/aicluster.py

```diff
@@ -0,0 +1,200 @@
+            if not metric_filename in index:
+                index[metric_filename] = {}
+                index[metric_filename]["uploaded-scribe"] = False
+        return index
+    index_key = s3.exists(prefix=benchmark_name, file_name=INDEX_FILE_NAME)
+    index = gen_index_obj(index_key)
+    metric_files = filter_metric_files(s3.list_directory(directory=None))
+    updated_index = update_index_from_metrics(index, metric_files)
+    return updated_index
+
+def upload_metrics_to_scribe(s3, benchmark_name, index, work_dir):
+    """
+    for each 'uploaded-scrbe: False' file in index
+      1. download it from S3
+      2. upload it to scribe
+      3. if success, update the index file with 'uploaded-scribe: True'
+    upload the updated index file to S3 after processing all files
+    """
+    try:
+        for index_key in index:
+            assert "uploaded-scribe" in index[index_key], \
+                f"Index key {index_key} missing field uploaded-scribe!"
+        index_file_path = work_dir.joinpath(INDEX_FILE_NAME)
+        with open(index_file_path, "w") as index_file:
+            yaml.safe_dump(index, index_file)
+        need_upload_metrics = filter(lambda x: not index[x]["uploaded-scribe"], index.keys())
+        for upload_metrics in need_upload_metrics:
+            # download the metrics file from S3 to work_dir
+            print(f"Downloading metrics file {upload_metrics} to local.")
+            metrics_key = s3.exists(prefix=None, file_name=upload_metrics)
+            assert metrics_key, f"Expected metrics file {upload_metrics} does not exist."
+            s3.download_file(metrics_key, work_dir)
+            # upload it to scribe
+            print(f"Uploading metrics file {upload_metrics} to scribe.")
+            metrics_path = str(work_dir.joinpath(upload_metrics).resolve())
+            with open(metrics_path, "r") as mp:
+                benchmark_time, benchmark_data = process_benchmark_json(mp)
+            uploader = TorchBenchUserbenchmarkUploader()
+            # user who run the benchmark on ai cluster
+            uploader.UNIX_USER = "diegosarina"
+            uploader.SUBMISSION_GROUP_GUID = "ai-cluster"
+            uploader.post_userbenchmark_results(benchmark_time, benchmark_data)
+            # update the index file
+            index[upload_metrics]["uploaded-scribe"] = True
+            with open(index_file_path, "w") as index_file:
+                yaml.safe_dump(index, index_file)
+    except subprocess.SubprocessError:
+        print(f"Failed to upload the file to scribe.")
+    finally:
+        # upload the result index file to S3
+        s3.upload_file(prefix=benchmark_name, file_path=index_file_path)
+
+def get_work_dir(benchmark_name):
+    workdir = Path(REPO_ROOT).joinpath(".userbenchmark").joinpath(benchmark_name).joinpath("logs")
+    workdir.mkdir(parents=True, exist_ok=True)
+    return workdir
+
+def run_aicluster_benchmark(benchmark_name: str, check_success=True, upload_scribe=True):
```

Comment:
Would I be correct that this function does not actually "run" the benchmark, instead it checks the result and upload accordingly?

Reply:
Correct, the workflow is:
1. every 8 PM UTC, the AI cluster will run a cron job that runs the distributed userbenchmark
2. if the cron job is successful, it will sync the result json files to Amazon S3 bucket
3. every 10 PM UTC, we will run this github actions workflow, and inspect into the S3 bucket to see if any new json file is there, if there is, add it to github action artifacts, and upload the json to the internal dashboard.